### PR TITLE
Component Gov round 7

### PR DESCRIPTION
**Pull request by jchen351**
### Description
  ignoreDirectories does not recursively include sub folders like we thought it would. We need to add additional sub folders.

### Motivation and Context
Fix CG :
1. https://aiinfra.visualstudio.com/Lotus/_componentGovernance/218239/alert/11474679?typeId=25427568
2. https://aiinfra.visualstudio.com/Lotus/_componentGovernance/218239/alert/11475140?typeId=25421034&pipelinesTrackingFilter=0

### CODE REVIEW

**Review by snnn — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/templates/component-governance-component-detection-steps.yml [hunk 1]
```diff
@@ -20,17 +20,20 @@ steps:
...
+      # ignore js node_modules directory. js node_modules directory is generated by js tests
       # ignore onnxruntime-inference-examples directory. onnxruntime-inference-examples directory is used for inference examples, not for onnxruntime package
       # ignore BinariesDirectory. BinariesDirectory is used for build output, not for onnxruntime package
       ignoreDirectories:
         '$(Build.Repository.LocalPath)/cmake/external/emsdk/upstream/emscripten/tests,
         $(Build.Repository.LocalPath)/cmake/external/onnx/third_party/benchmark,
         $(Build.Repository.LocalPath)/cmake/external/onnx/third_party/pybind11,
+        $(Build.Repository.LocalPath)/cmake/external/onnx/third_party/pybind11/tests,
         $(Build.Repository.LocalPath)/cmake/external/onnxruntime-extensions,
         $(Build.Repository.LocalPath)/js/react_native/e2e/node_modules,
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/templates/component-governance-component-detection-steps.yml:36]: Why the node_modules are ignored?

Comment (jchen351) [reply tools/ci_build/github/azure-pipelines/templates/component-governance-component-detection-steps.yml:36]: They are not part of the source code. They are generated.

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/component-governance-component-detection-steps.yml:36]: Generated by what? Should we monitor what nodejs modules are used in our pipelines?

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/component-governance-component-detection-steps.yml:36]: Are these files get distributed with our web and React Native packages?

Comment (jchen351) [reply tools/ci_build/github/azure-pipelines/templates/component-governance-component-detection-steps.yml:36]: No, from Yulong

**Pull request merged**